### PR TITLE
Use unique identifier to manipulate kube entity

### LINKED ISSUE #772: 2.0: Deleting a cluster removes a similarly named cluster from the SG db
Summary of Issue:
When deleting a cluster that has a name similar to an existing cluster, the similar name cluster is deleted from the SG db

Steps to Reproduce:
1. Create a cluster named eldridge-do (any size)
2. Create another cluster name eldridge-do1
3. After clusters are created, delete the eldridge-do cluster
Expected Results:
The cluster you selected is deleted. Not interacting with any other clusters.

Actual Results:
The cluster you deleted is deleted from the sg db and from DO. However, the eldridge-do1 cluster is also removed from the UI. The assets for the eldridge-do1 cluster is still in DO

Dev Info: (fill out and add links to log files)
1. `go version` : 1.10
2. SG latest commit hash or release tag: dddedff38267d34e16763fde1955ca4dec1cc5d7
3. Number of Masters and Nodes: Any
4. cloud provider: DO

### PR DESCRIPTION
**Pull request by stgleb**
ATTENTION, BREAKING CHANGE

Two separate installations of SG using single cloud account can affect each other
deleting kube with the same this needs introduction of ID for kube.
Now KubeID is used everywhere KubeName was used before.
KubeID is first 8 characters of cluster task id example:

```
{"clusterId":"da0b09e8","tasks":{"cluster":["da0b09e8-7589-4608-b56c-484b57126242"],"master":["bd978bb2-a9dc-4fc2-bb09-9bd90d8751fe"],"node":["607e8ef8-05e2-47aa-a9bd-cf72d34585b5"]}}
```

Kube structure

```
{  
   "id":"04438565",
   "state":"provisioning",
   "name":"gleb",
   "rbacEnabled":false,
   "accountName":"do",
   "region":"fra1",
   "apiPort":"",
   "auth":{  
      "username":"",
      "token":"",
      "caKey":"",
      "caCert":"",
      "cert":"",
      "key":""
   },
   "sshUser":"root",
   "sshKey":"",
   "operatingSystemVersion":"xenial",
   "dockerVersion":"17.06.0",
   "K8SVersion":"1.11.1",
   "helmVersion":"2.11.0",
   "networking":{  
      "manager":"0.10.0",
      "version":"0.10.0",
      "type":"vxlan",
      "cidr":"10.2.0.0/16"
   },
   "masters":{  
      "gleb-master-31d6":{  
         "id":"",
         "taskId":"31d6f658-3501-447d-bc1b-d6bbb47b90db",
         "role":"master",
         "createdAt":0,
         "provider":"digitalocean",
         "region":"fra1",
         "size":"s-1vcpu-2gb",
         "publicIp":"",
         "privateIp":"",
         "state":"building",
         "name":"gleb-master-31d6"
      }
   },
   "nodes":{  
      "gleb-node-5ec9":{  
         "id":"",
         "taskId":"5ec991b6-12af-433e-ba44-024bbadb6056",
         "role":"",
         "createdAt":0,
         "provider":"digitalocean",
         "region":"fra1",
         "size":"s-1vcpu-2gb",
         "publicIp":"",
         "privateIp":"",
         "state":"planned",
         "name":"gleb-node-5ec9"
      }
   }
}
```
Closes #772

### CODE REVIEW

**Comment by josephkane** [comment]
@stgleb will this change any api params at all? e.g. instead of GET /v1/api/kubes/{kubeName}, is it now GET /v1/api/kubes/{kubeID}?

File: pkg/kube/handler.go [hunk 1]
```diff
@@ -549,7 +549,7 @@ func (h *Handler) deleteNode(w http.ResponseWriter, r *http.Request) {
 	}
 
 	config := &steps.Config{
-		ClusterName:      k.Name,
+		ClusterID:        k.Name,
```
Comment (saheienko) [inline pkg/kube/handler.go]: Should it be `ClusterName`?

File: pkg/kube/service.go [hunk 2]
```diff
@@ -77,12 +78,16 @@ func NewService(prefix string, s storage.Interface, chrtGetter ChartGetter) *Ser
 
 // Create and stores a kube in the provided storage.
 func (s *Service) Create(ctx context.Context, k *model.Kube) error {
+	if k.ID == "" {
+		k.ID = uuid.New()[:8]
```
Comment (saheienko) [inline pkg/kube/service.go]: What if I try to create a few clusters with the same id?

Comment (stgleb) [reply pkg/kube/service.go]: @saheienko currenyly you can create a cluster from UI only through provisioner and it sets id to cluster

File: pkg/kube/service.go [hunk 3]
```diff
@@ -175,8 +180,8 @@ func (s *Service) GetKubeResources(ctx context.Context, kname, resource, ns, nam
 	}
 
 	req := client.Get().Resource(resource).Namespace(ns)
-	if name != "" {
-		req.Name(name)
+	if kubeID != "" {
+		req.Name(kubeID)
```
Comment (saheienko) [inline pkg/kube/service.go]: That's not a cluster name.

File: pkg/provisioner/provisioner.go [hunk 4]
```diff
@@ -53,13 +53,18 @@ func NewProvisioner(repository storage.Interface, kubeService KubeService) *Task
...
-func (tp *TaskProvisioner) ProvisionCluster(ctx context.Context, profile *profile.Profile, config *steps.Config) (map[string][]*workflows.Task, error) {
+// ProvisionCluster runs provisionCluster process among nodes
+// that have been provided for provisionCluster
+func (tp *TaskProvisioner) ProvisionCluster(ctx context.Context,
+	profile *profile.Profile, config *steps.Config) (map[string][]*workflows.Task, error) {
 	masterTasks, nodeTasks, clusterTask := tp.prepare(config.Provider, len(profile.MasterProfiles),
 		len(profile.NodesProfiles))
 
+	// Get clusterID from taskID
+	config.ClusterID = clusterTask.ID[:8]
```
Comment (saheienko) [inline pkg/provisioner/provisioner.go]: Please, check length of the string.

File: pkg/kube/service.go (hunk 2 above)
Comment (saheienko) [reply pkg/kube/service.go]: You can create it via the api.

**Pull request merged**